program/tests/e2e.rs

```diff
@@ -244,204 +241,57 @@ async fn test_e2e() {
-        check_token_account_balance(
-            &mut context,
-            &alice_token_account,
-            alice_token_account_starting_token_balance,
-        )
-        .await;
-        check_token_account_balance(&mut context, &escrow_token_account, 0).await;
     }
 
-    // Create a lockup for Bob.
-    let bob_lockup = Keypair::new();
-    let bob_lockup_amount = 2_000;
-    let bob_lockup_period_seconds = 120;
-    {
-        let clock = context
-            .banks_client
-            .get_sysvar::<Clock>()
-            .await
-            .expect("get_sysvar");
-
-        let rent = context.banks_client.get_rent().await.expect("get_rent");
-        let space = std::mem::size_of::<Lockup>();
-
-        send_transaction(
-            &mut context,
-            &[
-                system_instruction::transfer(
-                    &payer.pubkey(),
-                    &bob_lockup.pubkey(),
-                    rent.minimum_balance(space),
-                ),
-                system_instruction::allocate(&bob_lockup.pubkey(), space as u64),
-                system_instruction::assign(&bob_lockup.pubkey(), &paladin_lockup_program::id()),
-                paladin_lockup_program::instruction::lockup(
-                    &bob.pubkey(),
-                    &bob.pubkey(),
-                    &bob_token_account,
-                    &bob_lockup.pubkey(),
-                    &mint,
-                    bob_lockup_amount,
-                    bob_lockup_period_seconds,
-                    &spl_token_2022::id(),
-                ),
-            ],
-            &[&payer, &bob, &bob_lockup],
-        )
-        .await;
-
-        let expected_lockup_start = clock.unix_timestamp as u64;
-        let expected_lockup_end = expected_lockup_start.saturating_add(bob_lockup_period_seconds);
-
-        // Validate the lockup was created and tokens were transferred to the escrow.
-        check_lockup_state(
-            &mut context,
-            &bob_lockup.pubkey(),
-            &Lockup::new(
-                bob_lockup_amount,
-                &bob.pubkey(),
-                expected_lockup_start,
-                expected_lockup_end,
-                &mint,
-            ),
-        )
-        .await;
-        check_token_account_balance(
-            &mut context,
-            &bob_token_account,
-            bob_token_account_starting_token_balance.saturating_sub(bob_lockup_amount),
-        )
-        .await;
-        check_token_account_balance(&mut context, &escrow_token_account, bob_lockup_amount).await;
-    }
-
-    // Warp the clock 60 seconds.
-    // Bob can't withdraw until the period ends.
+    // Warp the clock 30 more minutes.
+    // Alice can now withdraw.
     {
         let mut clock = context
             .banks_client
             .get_sysvar::<Clock>()
             .await
             .expect("get_sysvar");
-        clock.unix_timestamp = clock.unix_timestamp.saturating_add(60);
+        clock.unix_timestamp = clock
```

Comment:
Wonder if here you can use your nice helper:
```rust
set_clock(
    &mut context,
    clock.unix_timestamp + LOCKUP_COOLDOWN_SECONDS as i64,
)
.await;
```

Reply:
I was being lazy :sweat_smile: I changed the function to be more generally useful by incrementing the clock instead of setting it, so it could be more easily reused